### PR TITLE
Fixes #214 : Provide Modulemd.Buildopts.equals() method

### LINKED ISSUE #214: Provide Modulemd.Buildopts.equals() method
In order to implement #186, we need an equality check for `Modulemd.Buildopts`. It should compare the `rpm_macros` string and the `whitelist` hash table.

### PR DESCRIPTION
**Pull request by OrionStar25**
Fixes #214

### CODE REVIEW

**Comment by OrionStar25** [comment]
@sgallagher  Please review

**Comment by sgallagher** [comment]
For the record, this code looks like it would work fine and if it was the only such instance, I'd merge it as-is. However, I'd prefer to get that requested common routine in first.

**Comment by OrionStar25** [comment]
@sgallagher added some tests here too.

File: modulemd/v2/modulemd-buildopts.c [hunk 1]
```diff
@@ -45,6 +45,36 @@ enum
...
+    return FALSE;
+
+  g_return_val_if_fail (MODULEMD_IS_BUILDOPTS (self_1), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_BUILDOPTS (self_2), FALSE);
+
+  if (!modulemd_hash_table_sets_are_equal (self_1->whitelist,
+                                           self_2->whitelist))
+    return FALSE;
+
+  if (!self_1->rpm_macros && !self_2->rpm_macros)
```
Comment (sgallagher) [inline modulemd/v2/modulemd-buildopts.c]: You don't need to do these NULL-checks because `g_strcmp0()` is NULL-aware. It will do this for you.

**Comment by OrionStar25** [comment]
@sgallagher Done! :fireworks:

**Pull request merged**